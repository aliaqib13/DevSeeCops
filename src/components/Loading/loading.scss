.loading-container {
  min-height: inherit;
  margin: -20px;
  display: flex;
  align-items: center;
  justify-content: center
}