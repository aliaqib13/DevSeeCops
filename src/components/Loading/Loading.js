import React from 'react';
import { Spin } from 'antd';
import './loading.scss';

const Loading = () => (
    <div className="loading-container">
        <Spin size="large" tip="Loading..." />
    </div>
);

export default Loading;
