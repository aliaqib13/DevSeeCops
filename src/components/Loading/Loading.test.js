import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from './Loading';

describe('Loading', () => {
    let component;

    beforeEach(() => {
        component = shallow(<Loading />);
    });

    it('Should render successfully', () => {
        const wrapper = component.find('.loading-container');
        expect(wrapper.exists()).toBeTruthy();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render Spin component', () => {
        const spin = component.find('Spin');
        expect(spin.exists()).toBeTruthy();
    });

    it('Should have tip "loading..."', () => {
        const spin = component.find('Spin');
        expect(spin.props().tip).toBe('Loading...');
    });
});
