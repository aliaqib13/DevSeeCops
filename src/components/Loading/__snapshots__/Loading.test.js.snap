// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Should render successfully 1`] = `
<div
  className="loading-container"
>
  <Spin
    size="large"
    spinning={true}
    tip="Loading..."
    wrapperClassName=""
  />
</div>
`;
